habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py

```diff
@@ -319,9 +319,23 @@ def updater(self):
     @property
     def policy_action_space(self):
         # TODO: Hack for discrete HL action spaces.
-        return spaces.MultiDiscrete(
-            tuple([agent.policy_action_space.n for agent in self._agents])
+        all_discreet = np.all(
+            [
+                isinstance(agent.policy_action_space, spaces.MultiDiscrete)
+                for agent in self._agents
+            ]
         )
+        if all_discreet:
+            return spaces.MultiDiscrete(
+                tuple([agent.policy_action_space.n for agent in self._agents])
+            )
+        else:
+            return spaces.Dict(
```

Comment:
I don't think Dict action spaces are supported by most of the code? But if this works then it's fine for now.

Reply:
I think this is only called here: https://github.com/facebookresearch/habitat-lab/blob/0317fc7cc9f3c93e3b9bf95b79837acdc3f24fca/habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py#L819

I updated the code so that that function can handle dicts.